woocommerce/woocommerce
# Fix @woocommerce/block-templates changes not being picked up on build

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





This PR addresses the following issue I was seeing when implementing #40620. I'm putting up this PR separately to keep the changes isolated and easily testable.

1. I built the monorepo using `pnpm run build` - no build errors
2. I made changes in `@woocommerce/block-templates`
3. I built the monorepo using `pnpm run build` - no build errors
4. Loading the new product editor screen, I did not see my changes
5. The `wc.blockTemplates` global also was not updated

Cleaning my build would result in my changes being reflected the first time but failing to be picked up when subsequent changes were then made to  `@woocommerce/block-templates`.



### How to test the changes in this Pull Request:



_**Note: The first time you build with this branch, you may get `@woocommerce/data` TypeScript errors. If this happens, running `git clean -fdx && pnpm install` should fix this. You should only have to run that once at most.**_

With a WooCommerce dev env with the new product editor enabled (**WooCommerce** > **Settings** > **Advanced** > **Features** > **Experimental features**)...

1. Make a change in `@woocommerce/block-templates`. I suggest adding a `console.log()` to `packages/js/block-templates/src/utils/register-woo-block-type.ts` in the `registerWooBlockType() function, since that is called by all of our blocks when the new product editor loads.

https://github.com/woocommerce/woocommerce/blob/bffb918958b766658e07dcede15fc06ba61ade1d/packages/js/block-templates/src/utils/register-woo-block-type.ts#L131

2. Run `pnpm run build`. Verify no build errors.
3. Reload **Products** > **Add New**. Verify changes made to `@woocommerce/block-templates` are reflected.
4. Make another change to `@woocommerce/block-templates`. I suggest just changing the console log ouput.
5. Run `pnpm run build`. Verify no build errors.
6. Reload **Products** > **Add New**. Verify changes made to `@woocommerce/block-templates` are reflected.



### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment 

</details>
